src/NimBLEService.cpp

```diff
@@ -159,7 +159,7 @@ bool NimBLEService::start() {
             // Nimble requires the last characteristic to have it's uuid = 0 to indicate the end
             // of the characteristics for the service. We create 1 extra and set it to null
             // for this purpose.
-            pChr_a = new ble_gatt_chr_def[numChrs + 1];
+            pChr_a = new ble_gatt_chr_def[numChrs + 1]{};
```

Comment:
Note that this uses "uniform initialization" with {}

This is a good way for modern C++ (C++11 and later)

If we're also targeting older C++, we should initialize using () instead.


For the different types of initialization and their pitfalls, refer to Scott Meyer https://ianyepan.github.io/posts/cpp-uniform-initialization/ or Herb Sutter https://herbsutter.com/2013/05/09/gotw-1-solution/